api/txspool.go

```diff
@@ -38,6 +40,79 @@ func (a *API) postPoolTx(c *gin.Context) {
 	c.JSON(http.StatusOK, writeTx.TxID.String())
 }
 
+func (a *API) postAtomicPool(c *gin.Context) {
+	// Parse body
+	var receivedTxs []receivedPoolTx
+	if err := c.ShouldBindJSON(&receivedTxs); err != nil {
+		retBadReq(err, c)
+		return
+	}
+	nTxs := len(receivedTxs)
+	if nTxs <= 1 {
+		retBadReq(errors.New(ErrSingleTxInAtomicEndpoint), c)
+		return
+	}
+	// Transform from received to insert format and validate (individually)
+	writeTxs := make([]l2db.PoolL2TxWrite, nTxs) // used for DB insert
+	txIDStrings := make([]string, nTxs)          // used for successful response
+	clientIP := c.ClientIP()
+	for i, tx := range receivedTxs {
+		writeTx := tx.toPoolL2TxWrite()
+		if err := a.verifyPoolL2TxWrite(writeTx); err != nil {
+			retBadReq(err, c)
+			return
+		}
+		writeTx.ClientIP = clientIP
+		writeTxs[i] = *writeTx
+		txIDStrings[i] = writeTx.TxID.String()
+	}
+	// Validate that all txs in the payload represent an atomic group
+	if !isAtomicGroup(writeTxs) {
+		retBadReq(errors.New(ErrTxsNotAtomic), c)
+		return
+	}
+	// Insert to DB
+	if err := a.l2.AddAtomicTxsAPI(writeTxs); err != nil {
+		retSQLErr(err, c)
+		return
+	}
+	// Return IDs of the added txs in the pool
+	c.JSON(http.StatusOK, txIDStrings)
+}
+
+// isAtomicGroup returns true if all the txs are needed to be forged
+// (all txs will be forged in the same batch or non of them will be forged)
+func isAtomicGroup(txs []l2db.PoolL2TxWrite) bool {
```

Comment:
I agree to use the graph package, but we can think about in use this nutshell implementation in the future:

https://github.com/hermeznetwork/hermez-node/blob/8c47d4f3a7b4dc3e46d23d16b0e8eb010a86144e/txselector/txbatch.go#L228-L293

Reply:
Can you create tests for this method?